### PR TITLE
Fix Type Error

### LINKED ISSUE #29: TypeError: 'set' object is not subscriptable
argh. introduced a new bug!
```
Thu Feb 16 06:00:01 CST 2017
INFO: Loading config from /opt/maxpoint/tools/conda-mirror-config.yaml
INFO: config: {'whitelist': [{'name': 'system'}], 'blacklist': [{'license': '*agpl*'}, {'license': 'None'}, {'license': ''}, {'name': 'marshmallow', 'build': 'py35_0', 'version': '2.10.4'}]}
Log level set to INFO
Traceback (most recent call last):
  File "/opt/maxpoint/envs/conda-mirror/bin/conda-mirror", line 11, in <module>
    sys.exit(cli())
  File "/opt/maxpoint/envs/conda-mirror/lib/python3.5/site-packages/conda_mirror/conda_mirror.py", line 227, in cli
    args.platform, blacklist, whitelist)
  File "/opt/maxpoint/envs/conda-mirror/lib/python3.5/site-packages/conda_mirror/conda_mirror.py", line 493, in main
    _validate_packages(possible_packages_to_mirror, local_directory)
  File "/opt/maxpoint/envs/conda-mirror/lib/python3.5/site-packages/conda_mirror/conda_mirror.py", line 364, in _validate_packages
    package_metadata = package_repodata[package]
TypeError: 'set' object is not subscriptable
```

### PR DESCRIPTION
**Pull request by ericdill**
Closes #29

### CODE REVIEW

**Comment by ericdill** [comment]
code review please @marshall245 @parente

File: test/test_conda_mirror.py [hunk 1]
```diff
@@ -93,30 +97,11 @@ def test_cli(tmpdir, channel, platform, repodata):
         assert len(rd['info']) == len(disk_info)
         assert len(rd['packages']) == len(disk_packages)
 
-
-def test_handling_bad_package(tmpdir, repodata):
```
Comment (parente) [inline test/test_conda_mirror.py]: Do the assertions in the test that calls this new function cover the purpose of the assertions in the removed `test_handling_bad_package`?

Comment (ericdill) [inline test/test_conda_mirror.py]: Not entirely. Will fix that in a new commit

Comment (ericdill) [inline test/test_conda_mirror.py]: Alright updated. Final check @parente ?

File: test/test_conda_mirror.py [hunk 2]
```diff
@@ -71,6 +71,16 @@ def test_cli(tmpdir, channel, platform, repodata):
                          platform=platform)
     old_argv = copy.deepcopy(sys.argv)
     sys.argv = cli_args.split(' ')
+    # Write a package that does not exist in the upstreap repodata into the mirror path
+    # to make sure we exercise a broken code path
+    # https://github.com/maxpoint/conda-mirror/issues/29
+    _write_bad_package(channel_dir=f2.strpath, platform_name=platform,
+                       pkg_name='bad-1-0.tar.bz2')
+    # Write a bad package that does exist in the upstreap repodata into the mirror path
```
Comment (parente) [inline test/test_conda_mirror.py]: upstreap -> upstream

Comment (parente) [inline test/test_conda_mirror.py]: One typo above. Then :shipit:

Comment (ericdill) [inline test/test_conda_mirror.py]: Good catch. Thanks @parente

**Pull request merged**